### DISCUSSION TITLE
NameError:Undefined

### DISCUSSION THREAD

**Discussion by GuGuMur** [sqlalchemy/mako#374] [Usage Questions]
Hello!
I'm using Mako to generate wikitext.
I want:
* If `song_name2` in the parameters set: return the text;
* If not: no treatment.

My code:
```python
from mako.template import Template
temp = Template("""
% if song_name2:
|second=${song_name2}
% endif
"""))
temp.render(song_name2="Stasis")   #return "|second=Stasis"
temp.render()                      #return NameError:Undefined——I want an empty string.
```
Thanks!

**Comment by zzzeek** [comment]
the example as given works without issue, because you dont have [strict_undefined](https://docs.makotemplates.org/en/latest/usage.html#mako.template.Template.params.strict_undefined) configured:

```py
from mako.template import Template
temp = Template("""
% if song_name2:
|second=${song_name2}
% endif
""")
print(temp.render(song_name2="Stasis"))
print(temp.render())
```

if you do have strict_undefined configured, give the template a default using [`<%page>`](https://docs.makotemplates.org/en/latest/syntax.html#page):

```py
from mako.template import Template
temp = Template("""
<%page args="song_name2='default song'"/>

% if song_name2:
|second=${song_name2}
% endif
""", strict_undefined=True)
print(temp.render(song_name2="Stasis"))
print(temp.render())
```

**Reply by GuGuMur** [reply]
It works! Many thanks for your kind and warm help! It seems that I still need to learn more deeply about Mako lol
